syntax = "proto3";

package NAlice;

option go_package = "a.yandex-team.ru/alice/megamind/protos/common";
option java_package = "ru.yandex.alice.megamind.protos.common";

enum EDataSourceType {
    reserved 1, 3, 5, 18;
    reserved "BEGEMOT", "VINS_WIZARD", "BEGEMOT_TAGGER_RESULT", "PERSONAL_INTENTS";

    UNDEFINED_DATA_SOURCE = 0; // The first enum value must be zero in proto3.

    // BEGEMOT = 1;
    BLACK_BOX = 2;
    // VINS_WIZARD = 3;
    USER_LOCATION = 4;
    // BEGEMOT_TAGGER_RESULT = 5;
    BEGEMOT_EXTERNAL_MARKUP = 6;
    DIALOG_HISTORY = 7;
    BEGEMOT_IOT_NLU_RESULT = 8;
    GC_MEMORY_STATE = 9;
    BEGEMOT_ITEM_SELECTOR_RESULT = 11;
    WEB_SEARCH_DOCS = 12;
    WEB_SEARCH_DOCS_RIGHT = 13;
    WEB_SEARCH_WIZPLACES = 14;
    WEB_SEARCH_SUMMARIZATION = 15;
    WEB_SEARCH_RENDERRER = 16;
    ENTITY_SEARCH = 17;
    // PERSONAL_INTENTS = 18;
    VINS_WIZARD_RULES = 19;
    SMART_HOME_INFO = 20;
    VIDEO_VIEW_STATE = 21;
    NOTIFICATION_STATE = 22;
    ALICE4BUSINESS_DEVICE = 23;
    BEGEMOT_FIXLIST_RESULT = 24;
    WEB_SEARCH_WIZARD = 25;
    WEB_SEARCH_BANNER = 26;
    RESPONSE_HISTORY = 27;
    SKILL_DISCOVERY_GC = 28;

    DEVICE_STATE_NAVIGATOR = 29;

    IOT_USER_INFO = 30;
    APP_INFO = 31;

    EMPTY_DEVICE_STATE = 32; // If set, device state won't be provided in scenario request
    RAW_PERSONAL_DATA = 33;
    VIDEO_CURRENTLY_PLAYING = 34;
    CONTACTS_LIST = 35;
    ENVIRONMENT_STATE = 36;
    TANDEM_ENVIRONMENT_STATE = 37;

    BEGEMOT_BEGGINS_RESULT = 38; // Do not use it
    WEB_SEARCH_REQUEST_META = 39;

    WHISPER_INFO = 40;

    GUEST_DATA = 41;
    GUEST_OPTIONS = 42;

    QUASAR_DEVICES_INFO = 43;
}
